server/src/main/java/org/opensearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -441,7 +441,10 @@ private void onPhaseEnd(SearchRequestContext searchRequestContext) {
 
     private void onPhaseStart(SearchPhase phase) {
         setCurrentPhase(phase);
-        this.searchRequestContext.getSearchRequestOperationsListener().onPhaseStart(this);
+        try {
+            SearchPhaseName.valueOf(phase.getSearchPhaseName().toString());
+            this.searchRequestContext.getSearchRequestOperationsListener().onPhaseStart(this);
+        } catch (IllegalArgumentException ignored) {}
```

Comment:
Would be good to atleast capture this as part of trace logs.

Reply:
+1

Reply:
Not needed as try catch logic is removed